### ISSUE TITLE
Documentation not up to date

### ISSUE THREAD

**Issue by m-gora** [TinyModularThings/Chunk-Pregenerator-Issue-Tracker#9]
MC Version: 1.16
Mod Version 3.5.0

The initial generation worked with the command presented in the config. However neither the **pregen tasklist gen** or expansion etc worked. 
I assume that the functionality works, but the documentation is not up to date. 

It would also be more helpful if commands would give more hints how to use the command properly rather than just saying that it's improperly used.

**Comment by Speiger** [comment]
Oh yeah the documentation is setup for "advanced" commands which means all the "special" modifiers only work with it.
And the dummed down version doesn't support it.

I need to add a info that these special modifiers do not work in "simple command mode".
(Chunk Pregenerator is required on the client with advanced mode, which is not always wanted)

**Comment by m-gora** [comment]
@Speiger i'm running chunk pregenerator server side only. So can I use it in advanced mode or not? In 1.12 I was running pregen in worldborder mode as a nightly cronjob to generate chunks while the server is unpopulated and I want to configure it the same way for 1.16.

**Comment by Speiger** [comment]
@m-gora advanced mode would allow for this.
```
/pregen start gen radius Testing SQUARE ^ ^ b10000
```
while simple mode would be this:
```
/pregen start gen radius Testing SQUARE 0 0 625
```
The ^ is the one that isn't allowed in simple or the "b" modifier.

And what you are speaking about is the "Singleplayer required" bug.

Ignoring that you can pause tasks ingame and the player detector is in the config exlusively, but it updates instantly.

Otherwise i am not sure what you mean.

**Comment by m-gora** [comment]
Ok, one Issue I have with radius is the following line from the docs:
`Radius[Number]: The Radius of how large the Pregeneration Task should be. (Range: 1-300)`
which means the limit is 4800 blocks around the coordinates. 

I run an an adventuring modpack where players run around a lot, and in order to prevent live chunk generation, I want to run the pregenerator nighty. I created the first 100 chunks around our base by hand and then wanted to expand the area  each night. 

`pregen tasklist gen` also doeas not work, even though I remember it worked in 1.12. Since you don't have the project open source I can't even look up how it would work for sure and I need to rely completely on the docs with all caveats documented...

What is the singleplayer required bug now? I am not using the pregenerator in singleplayer...

**Comment by Speiger** [comment]
> Radius[Number]: The Radius of how large the Pregeneration Task should be. (Range: 1-300)
which means the limit is 4800 blocks around the coordinates.
>
>I run an an adventuring modpack where players run around a lot, and in order to prevent live chunk generation, I want to run the pregenerator nighty. I created the first 100 chunks around our base by hand and then wanted to expand the area each night.

The documentation is not updated with the 300 chunk radius is no longer accurate.
The limit is now 500 chunks (so 8k Block Radius)

But that is a "per task" limit, not a global limit.
And Expansions can stretch it a bit. Since that limit is 500 chunk radius worth of chunks.

If you need larger tasks sizes at ones, that isn't free anymore.

> pregen tasklist gen also doeas not work, even though I remember it worked in 1.12. Since you don't have the project open source I can't even look up how it would work for sure and I need to rely completely on the docs with all caveats documented...

Chunk pregen ingame has full autocompletion support. Some server consoles support that others not.
The documentation is for help, and yes it needs to be reworked, but you will have a lot nicer experience when you are doing this ingame instead of a server console because then you can see all commands as they are designed.


> What is the singleplayer required bug now? I am not using the pregenerator in singleplayer...

Basically if you play on MP server you need to have the mod installed on the client too to actually join the server.
The bug is, that you actually do not need it on the client when you join a server with said mod installed.
I am fixing that hopefully this week, with a few other things i need to address.

I was just swamped with ic2c since that is a 1000x larger mod that i wanted to be done with.

**Comment by Speiger** [comment]
Small note. 
I am doing a bugfix patch at the moment.
Once that is done i am going to redo the documentation.

It will take a bit of time, but I will do it over time.
I am known for being slow but steady.

**Comment by Speiger** [comment]
@m-gora documentation got updated with a redesign on everything 1.14.4 or newer.

Took a while because i avoided it for a long time, but now its done and 100x better!
